### PR TITLE
Feature/drive conformance

### PR DESCRIPTION
**Pull request by andrewkrug**
* support renaming teamDrives in interactive mode
* enhance logging
* deliver logs to consolidated stream
* disallow reuse of drive names unless cleared from state table
* add state table
* add drive conformance feature
* filter grouplist by prefix at origin

### CODE REVIEW

File: gsuite_driver/main.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                interactive_mode=driver_mode
+            )
+
+            conformant_name = this_drive._name_conformance(
+                drive_name=drive.get('name')
+            )
+
+            this_drive.update(drive.get('id'), conformant_name)
+
+        return 'Completed conformance mode pass on the teamDrives.'
```
Comment (gdestuynder) [inline gsuite_driver/main.py]: informational string on return feels strange

File: gsuite_driver/main.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    logger.debug('Searching DynamoDb for people.')
+    people = People()
+
+    logger.debug('Filtering person list to groups.')
+    groups = people.grouplist(filter=filter_prefix)
```
Comment (gdestuynder) [inline gsuite_driver/main.py]: `filter` is a reserved python class (global scope), normally, this should not be used as a variable name as it overwrites the class

File: gsuite_driver/main.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+        community_drive_driver.drive_name=drive_name
+
+        try:
+            community_drive_driver.find_or_create()
+            work_plan = community_drive_driver.reconcile_members(people.build_email_list(group))
+            community_drive_driver.execute_proposal(work_plan)
+        except DriveNameLockedError:
+            logger.warn('Skipping drive due to locked name: {}'.format(drive_name))
+    return 'Completed execution of teamDrive connector.'
```
Comment (gdestuynder) [inline gsuite_driver/main.py]: ^^ info string on return

File: gsuite_driver/serverless.yml [hunk 4]
```diff
@@ -49,15 +50,47 @@ provider:
...
+      Type: AWS::DynamoDB::Table
+      Properties:
+        TableName: gsuite-driver-state
+        AttributeDefinitions:
+          - AttributeName: name
+            AttributeType: S
+        KeySchema:
+          - AttributeName: name
+            KeyType: HASH
+        ProvisionedThroughput:
```
Comment (gdestuynder) [inline gsuite_driver/serverless.yml]: should this autoscale?

File: gsuite_driver/vault.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
+import boto3
```
Comment (gdestuynder) [inline gsuite_driver/vault.py]: Not calling Person-API?

File: gsuite_driver/vault.py [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+        return unique_groups
+
+    def build_email_list(self, group_dict):
+        memberships = []
+        for member in group_dict['members']:
+            if member.get('primaryEmail').split('@')[1] == 'mozilla.com':
+                memberships.append(member.get('primaryEmail'))
+                continue
+
+            if member.get('primaryEmail').split('@')[1] == 'mozillafoundation.org':
```
Comment (gdestuynder) [inline gsuite_driver/vault.py]: This is missing `getpocket.com` - though Ideally this should use HRIS data instead

File: tests/fake_cis.py [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+            'groups': self.groups(),
+            'primaryEmail': self.primary_email(),
+            'emails': self.additional_emails(),
+            'active': random.choice(active)
+        }
+
+        return profile
+
+    def user_id(self):
+        providers = ['ad|Mozilla-LDAP|', 'github|', 'google-oauth2|', 'email|']
```
Comment (gdestuynder) [inline tests/fake_cis.py]: Note if you want we also have `ad|Mozilla-LDAP-Dev` though i believe these are enough ;-)

File: gsuite_driver/vault.py [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+                memberships.append(member.get('primaryEmail'))
+                continue
+
+            for email in member['emails']:
```
Comment (gdestuynder) [inline gsuite_driver/vault.py]: XXX reminder that with profile v2 this will break

File: gsuite_driver/main.py (hunk 1 above)
Comment (andrewkrug) [reply gsuite_driver/main.py]: This return is what end up in cloudwatch.  I am open to standardizing this as some kind of summary statistic for all cis services.

File: gsuite_driver/main.py (hunk 2 above)
Comment (andrewkrug) [reply gsuite_driver/main.py]: Will fix this.

File: gsuite_driver/main.py (hunk 3 above)
Comment (andrewkrug) [reply gsuite_driver/main.py]: What kind of info would you like in the message?

File: gsuite_driver/serverless.yml (hunk 4 above)
Comment (andrewkrug) [reply gsuite_driver/serverless.yml]: Doesn't need to autoScale as this is the singular only batch based consumer.

File: gsuite_driver/vault.py (hunk 5 above)
Comment (andrewkrug) [reply gsuite_driver/vault.py]: Not calling personAPI until personAPI has a batch feature.

File: gsuite_driver/vault.py (hunk 6 above)
Comment (andrewkrug) [reply gsuite_driver/vault.py]: Will fix this.

File: gsuite_driver/vault.py (hunk 8 above)
Comment (andrewkrug) [reply gsuite_driver/vault.py]: Fa sho.  Will refactor at that time.  We'll probably end up with 2 versions of the IDVault whilst in transition.

**Comment by andrewkrug** [comment]
@gdestuynder see my comments above.  I'm totally willing to do another pass on this for the willfix items.

**Comment by gdestuynder** [comment]
you fixed the thing i blocked on (getpocket) so we're good
i dont know what the best return to cloud watch is, it just isnt a pattern i normally see - but functions dont normally return to a logger

**Pull request merged**